WalletWasabi.Tests/UnitTests/Blockchain/TransactionBuilding/ChangelessTransactionCoinSelectorTests.cs

```diff
@@ -0,0 +1,58 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Blockchain.Analysis.Clustering;
+using WalletWasabi.Blockchain.Keys;
+using WalletWasabi.Blockchain.TransactionBuilding;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Blockchain.Transactions;
+using WalletWasabi.Models;
+using WalletWasabi.Tests.Helpers;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.Blockchain.TransactionBuilding;
+
+/// <summary>
+/// Tests for <see cref="ChangelessTransactionCoinSelector"/> class.
+/// </summary>
+public class ChangelessTransactionCoinSelectorTests
+{
+	[Fact]
+	public void FindBestSolution()
+	{
+		using Key key = new();
+
+		List<SmartCoin> coins = GenerateDummySmartCoins(key, 6_025, 6_561, 8_192, 13_122, 50_000, 100_000, 196_939, 524_288);
+		long target = 150_000;
+
+		bool found = ChangelessTransactionCoinSelector.TryGetCoins(coins, new FeeRate(satoshiPerByte: 4), target, out List<SmartCoin>? selectedCoins);
+		Assert.True(found);
+
+		long[] solution = selectedCoins!.Select(x => x.Amount.Satoshi).ToArray();
+		Assert.Equal(new long[] { 100_000, 50_000, 6_025 }, solution);
+		Assert.Equal(6025, solution.Sum() - target); // ~4% more for the privacy.
+	}
+
+	private List<SmartCoin> GenerateDummySmartCoins(Key key, params long[] values)
+	{
+		Network network = Network.Main;
```

Comment:
Just curious, why `Network.Main`?

Reply:
It can be whatever network but it probably makes sense to use in tests `Network.Main` to simulate as much as possible what users actually utilize.

Reply:
Makes sense, thanks!